client/components/domains/registrant-extra-info/ca-form.jsx

```diff
@@ -138,30 +137,32 @@ export class RegistrantExtraInfoCaForm extends React.PureComponent {
 	};
 
 	handleChangeEvent = ( event ) => {
-		const { value, name, checked, type, id } = event.target;
+		const { value, checked, type, id } = event.target;
 		const newContactDetails = {};
 
-		if ( name === 'organization' ) {
-			newContactDetails[ name ] = value;
-			this.validateContactDetails( {
-				...this.props.contactDetails,
-				[ name ]: value,
+		if ( id === 'legal-type' ) {
+			this.props.updateRequiredDomainFields( {
+				organization: this.isCorporationLegalType( value ),
 			} );
-		} else {
-			newContactDetails.extra = {
-				ca: { [ camelCase( id ) ]: type === 'checkbox' ? checked : value },
-			};
 		}
 
+		newContactDetails.extra = {
+			ca: { [ camelCase( id ) ]: type === 'checkbox' ? checked : value },
+		};
+
 		this.props.updateContactDetailsCache( { ...newContactDetails } );
 
 		if ( this.props.isManaged ) {
 			this.props.onContactDetailsChange?.( { ...newContactDetails } );
 		}
 	};
 
+	isCorporationLegalType( legalType ) {
+		return legalType === 'CCO';
+	}
+
 	needsOrganization() {
-		return get( this.props.ccTldDetails, 'legalType' ) === 'CCO';
+		return this.isCorporationLegalType( get( this.props.ccTldDetails, 'legalType' ) );
```

Comment:
Minor, since I know this was pre-existing, but `get( this.props.ccTldDetails, 'legalType' )` could be `this.props.ccTldDetails?.legalType` without needing lodash.